File: C++/Vehicle With Polymorphism/Vehicle With Polymorphism/Truck.cpp
#include "Truck.h"
#include <iostream>

Truck::Truck(const std::string& make, const std::string& model, int year) : Vehicle(make, model, year) {}

void Truck::drive() {
	std::cout << "Driving a truck...\n";
}

void Truck::pullTrailer() {
	std::cout << "Pulling a trailer...\n";
}

void Truck::honk() {
	std::cout << "AHHHHHH honk!\n";
}

File: 7 - C# and ASP.NET/Assignments/Dog Class/Car (Maya's Code)/Car/Truck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Car
{
    internal class Truck : Vehicle
    {
        public int NumWheels { get; set; }
        public Truck(string make, string model, int year, int numWheels) : base(make, model, year)
        {
            this.NumWheels = numWheels;
        }
        public override string Drive()
        {
            return "Truck is driving.";
        }
        public void Haul()
        {
            MessageBox.Show("Truck is hauling.");
        }
    }
}
